### PR TITLE
Update manipulating-the-dom-with-refs.md

### PR DESCRIPTION
**Pull request by MadDevMax**
translated manipulating dom page.

### CODE REVIEW

File: src/content/learn/manipulating-the-dom-with-refs.md [hunk 1]
```diff
@@ -1,52 +1,52 @@
 ---
-title: 'Manipulating the DOM with Refs'
+title: 'Refs के साथ DOM में बदलाव करना'
 ---
 
 <Intro>
 
-React automatically updates the [DOM](https://developer.mozilla.org/docs/Web/API/Document_Object_Model/Introduction) to match your render output, so your components won't often need to manipulate it. However, sometimes you might need access to the DOM elements managed by React--for example, to focus a node, scroll to it, or measure its size and position. There is no built-in way to do those things in React, so you will need a *ref* to the DOM node.
+React स्वचालित रूप से [DOM](https://developer.mozilla.org/docs/Web/API/Document_Object_Model/Introduction) को आपके रेंडर आउटपुट के अनुसार अपडेट करता है, जिससे आपके कंपोनेंट्स को अक्सर इसे मैन्युपुलेट करने की आवश्यकता नहीं होती। हालांकि, कभी-कभी आपको React द्वारा मैनेज किए गए DOM एलिमेंट्स तक पहुंचने की आवश्यकता हो सकती है—जैसे किसी नोड को फोकस करना, स्क्रॉल करना, या उसका आकार और स्थिति मापना। React में इन चीज़ों के लिए कोई बिल्ट-इन तरीका नहीं है, इसलिए आपको DOM नोड के लिए एक *ref* की आवश्यकता होगी।
```
Comment (arshadkazmi42) [inline src/content/learn/manipulating-the-dom-with-refs.md:7]:
```suggestion
React स्वचालित रूप से [DOM](https://developer.mozilla.org/docs/Web/API/Document_Object_Model/Introduction) को आपके रेंडर आउटपुट के अनुसार अपडेट करता है, जिससे आपके कौम्पोनॅन्टस को अक्सर इसे मैन्युपुलेट करने की आवश्यकता नहीं होती। हालांकि, कभी-कभी आपको React द्वारा मैनेज किए गए DOM एलिमेंट्स तक पहुंचने की आवश्यकता हो सकती है—जैसे किसी नोड को फोकस करना, स्क्रॉल करना, या उसका आकार और स्थिति मापना। React में इन चीज़ों के लिए कोई बिल्ट-इन तरीका नहीं है, इसलिए आपको DOM नोड के लिए एक *ref* की आवश्यकता होगी।
```

we prefer using `कौम्पोनॅन्टस` as translation for components. can you update whole file to use this for components.

File: src/content/learn/manipulating-the-dom-with-refs.md [hunk 2]
```diff
@@ -1,52 +1,52 @@
...
 
 <YouWillLearn>
 
-- How to access a DOM node managed by React with the `ref` attribute
-- How the `ref` JSX attribute relates to the `useRef` Hook
-- How to access another component's DOM node
-- In which cases it's safe to modify the DOM managed by React
+- React द्वारा मैनेज किए गए DOM नोड तक `ref` एट्रिब्यूट के साथ कैसे पहुंचें
+- `ref` JSX एट्रिब्यूट का `useRef` हुक से क्या संबंध है
+- किसी दूसरे कंपोनेंट के DOM नोड तक कैसे पहुंचें
```
Comment (arshadkazmi42) [inline src/content/learn/manipulating-the-dom-with-refs.md:15]:
```suggestion
- किसी दूसरे कौम्पोनॅन्ट के DOM नोड तक कैसे पहुंचें
```

File: src/content/learn/manipulating-the-dom-with-refs.md [hunk 3]
```diff
@@ -1,52 +1,52 @@
...
 
-To access a DOM node managed by React, first, import the `useRef` Hook:
+React द्वारा मैनेज किए गए DOM नोड तक पहुंचने के लिए, सबसे पहले `useRef` हुक को इम्पोर्ट करें:
 
 ```js
 import { useRef } from 'react';
 ```
 
-Then, use it to declare a ref inside your component:
+फिर, इसे अपने कंपोनेंट के अंदर एक ref घोषित करने के लिए उपयोग करें:
```
Comment (arshadkazmi42) [inline src/content/learn/manipulating-the-dom-with-refs.md:28]:
```suggestion
फिर, इसे अपने कौम्पोनॅन्ट के अंदर एक ref डिक्लेअर करने के लिए उपयोग करें:
```

File: src/content/learn/manipulating-the-dom-with-refs.md [hunk 4]
```diff
@@ -1,52 +1,52 @@
...
 
-Finally, pass your ref as the `ref` attribute to the JSX tag for which you want to get the DOM node:
+अंत में, अपने ref को उस JSX टैग के `ref` एट्रिब्यूट के रूप में पास करें जिसके लिए आप DOM नोड प्राप्त करना चाहते हैं:
 
 ```js
 <div ref={myRef}>
 ```
 
-The `useRef` Hook returns an object with a single property called `current`. Initially, `myRef.current` will be `null`. When React creates a DOM node for this `<div>`, React will put a reference to this node into `myRef.current`. You can then access this DOM node from your [event handlers](/learn/responding-to-events) and use the built-in [browser APIs](https://developer.mozilla.org/docs/Web/API/Element) defined on it.
+`useRef` हुक एक ऑब्जेक्ट रिटर्न करता है जिसमें एकमात्र प्रॉपर्टी होती है, जिसे `current` कहा जाता है। शुरुआत में, `myRef.current` का मान `null` होगा। जब React इस `<div>` के लिए एक DOM नोड बनाएगा, तो React इस नोड का रेफरेंस `myRef.current` में डाल देगा। इसके बाद आप इस DOM नोड को अपने [इवेंट हैंडलर्स](/learn/responding-to-events) से एक्सेस कर सकते हैं और उस पर परिभाषित [ब्राउज़र APIs](https://developer.mozilla.org/docs/Web/API/Element) का उपयोग कर सकते हैं।
```
Comment (arshadkazmi42) [inline src/content/learn/manipulating-the-dom-with-refs.md:40]:
```suggestion
`useRef` हुक एक ऑब्जेक्ट रिटर्न करता है जिसमें एकमात्र प्रॉपर्टी होती है, जिसे `current` कहा जाता है। शुरुआत में, `myRef.current` का मान `null` होगा। जब React इस `<div>` के लिए एक DOM नोड बनाएगा, तो React इस नोड का रेफरेंस `myRef.current` में डाल देगा। इसके बाद आप इस DOM नोड को अपने [इवेंट हैंडलर्स](/learn/responding-to-events) से एक्सेस कर सकते हैं और उसकी बिल्ट-इन [ब्राउज़र APIs](https://developer.mozilla.org/docs/Web/API/Element) का उपयोग कर सकते हैं।
```

File: src/content/learn/manipulating-the-dom-with-refs.md [hunk 5]
```diff
@@ -73,18 +73,18 @@ export default function Form() {
...
 </Sandpack>
 
-To implement this:
+इसका उपयोग करने के लिए:
 
-1. Declare `inputRef` with the `useRef` Hook.
-2. Pass it as `<input ref={inputRef}>`. This tells React to **put this `<input>`'s DOM node into `inputRef.current`.**
-3. In the `handleClick` function, read the input DOM node from `inputRef.current` and call [`focus()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/focus) on it with `inputRef.current.focus()`.
-4. Pass the `handleClick` event handler to `<button>` with `onClick`.
+1. `useRef` हुक के साथ `inputRef` घोषित करें।
```
Comment (arshadkazmi42) [inline src/content/learn/manipulating-the-dom-with-refs.md:78]:
```suggestion
1. `useRef` हुक के साथ `inputRef` डिक्लेअर करें।  
```

File: src/content/learn/manipulating-the-dom-with-refs.md [hunk 6]
```diff
@@ -124,35 +124,35 @@ export default function CatFriends() {
     <>
       <nav>
         <button onClick={handleScrollToFirstCat}>
-          Neo
+          नियो
```
Comment (arshadkazmi42) [inline src/content/learn/manipulating-the-dom-with-refs.md:127]:
```suggestion
          Neo
```

Lets not translate these names

File: src/content/learn/manipulating-the-dom-with-refs.md [hunk 7]
```diff
@@ -124,35 +124,35 @@ export default function CatFriends() {
...
         </button>
         <button onClick={handleScrollToSecondCat}>
-          Millie
+          मिली
```
Comment (arshadkazmi42) [inline src/content/learn/manipulating-the-dom-with-refs.md:130]:
```suggestion
          Millie
```

Lets not translate these names

File: src/content/learn/manipulating-the-dom-with-refs.md [hunk 8]
```diff
@@ -124,35 +124,35 @@ export default function CatFriends() {
...
         </button>
         <button onClick={handleScrollToThirdCat}>
-          Bella
+          बेला
```
Comment (arshadkazmi42) [inline src/content/learn/manipulating-the-dom-with-refs.md:133]:
```suggestion
          Bella
```

Lets not translate these names

File: src/content/learn/manipulating-the-dom-with-refs.md [hunk 9]
```diff
@@ -124,35 +124,35 @@ export default function CatFriends() {
...
         </button>
       </nav>
       <div>
         <ul>
           <li>
             <img
               src="https://placecats.com/neo/300/200"
-              alt="Neo"
+              alt="नियो"
```
Comment (arshadkazmi42) [inline src/content/learn/manipulating-the-dom-with-refs.md:141]:
```suggestion
              alt="Neo"
```

File: src/content/learn/manipulating-the-dom-with-refs.md [hunk 10]
```diff
@@ -124,35 +124,35 @@ export default function CatFriends() {
...
               ref={firstCatRef}
             />
           </li>
           <li>
             <img
               src="https://placecats.com/millie/200/200"
-              alt="Millie"
+              alt="मिली"
```
Comment (arshadkazmi42) [inline src/content/learn/manipulating-the-dom-with-refs.md:148]:
```suggestion
              alt="Millie"
```

File: src/content/learn/manipulating-the-dom-with-refs.md [hunk 11]
```diff
@@ -124,35 +124,35 @@ export default function CatFriends() {
...
               ref={secondCatRef}
             />
           </li>
           <li>
             <img
               src="https://placecats.com/bella/199/200"
-              alt="Bella"
+              alt="बेला"
```
Comment (arshadkazmi42) [inline src/content/learn/manipulating-the-dom-with-refs.md:155]:
```suggestion
              alt="Bella"
```

**Review by arshadkazmi42 — CHANGES_REQUESTED** [review]
Looks good. I have left a few minor comments. Can you take a look at those. Post that we can get this merged.

File: src/content/learn/manipulating-the-dom-with-refs.md (hunk 1 above)
Comment (MadDevMax) [reply src/content/learn/manipulating-the-dom-with-refs.md]: Replaced in the whole file.

File: src/content/learn/manipulating-the-dom-with-refs.md (hunk 2 above)
Comment (MadDevMax) [reply src/content/learn/manipulating-the-dom-with-refs.md]: Replaced in the whole file.

File: src/content/learn/manipulating-the-dom-with-refs.md (hunk 3 above)
Comment (MadDevMax) [reply src/content/learn/manipulating-the-dom-with-refs.md]: Replaced in the whole file.

File: src/content/learn/manipulating-the-dom-with-refs.md (hunk 4 above)
Comment (MadDevMax) [reply src/content/learn/manipulating-the-dom-with-refs.md]: changed as suggested.

File: src/content/learn/manipulating-the-dom-with-refs.md (hunk 5 above)
Comment (MadDevMax) [reply src/content/learn/manipulating-the-dom-with-refs.md]: changed as suggested

File: src/content/learn/manipulating-the-dom-with-refs.md (hunk 6 above)
Comment (MadDevMax) [reply src/content/learn/manipulating-the-dom-with-refs.md]: reverted name transalation

File: src/content/learn/manipulating-the-dom-with-refs.md (hunk 7 above)
Comment (MadDevMax) [reply src/content/learn/manipulating-the-dom-with-refs.md]: reverted name transalation

File: src/content/learn/manipulating-the-dom-with-refs.md (hunk 8 above)
Comment (MadDevMax) [reply src/content/learn/manipulating-the-dom-with-refs.md]: reverted name transalation

File: src/content/learn/manipulating-the-dom-with-refs.md (hunk 9 above)
Comment (MadDevMax) [reply src/content/learn/manipulating-the-dom-with-refs.md]: reverted name transalation

File: src/content/learn/manipulating-the-dom-with-refs.md (hunk 10 above)
Comment (MadDevMax) [reply src/content/learn/manipulating-the-dom-with-refs.md]: reverted name transalation

File: src/content/learn/manipulating-the-dom-with-refs.md (hunk 11 above)
Comment (MadDevMax) [reply src/content/learn/manipulating-the-dom-with-refs.md]: reverted name transalation

File: src/content/learn/manipulating-the-dom-with-refs.md [hunk 12]
```diff
@@ -587,15 +589,15 @@ for (let i = 0; i < 20; i++) {
 
 </DeepDive>
 
-## Best practices for DOM manipulation with refs {/*best-practices-for-dom-manipulation-with-refs*/}
+## DOM मैनिपुलेशन के लिए सर्वोत्तम प्रथाएँ (Best practices for DOM manipulation with refs)
```
Comment (arshadkazmi42) [inline src/content/learn/manipulating-the-dom-with-refs.md:592]:
```suggestion
## DOM मैनिपुलेशन के लिए सर्वोत्तम प्रथाएँ (Best practices for DOM manipulation with refs) {/*best-practices-for-dom-manipulation-with-refs*/}
```

do not remove heading links which are there in `{}` due to this you were seeing the error while commit

File: src/content/learn/manipulating-the-dom-with-refs.md [hunk 13]
```diff
@@ -636,30 +638,30 @@ button {
...
+- यदि आप React द्वारा प्रबंधित DOM नोड्स को बदलते हैं, तो केवल उन हिस्सों को बदलें जिन्हें React अपडेट करने का कोई कारण नहीं है।
 
 </Recap>
 
 
 
 <Challenges>
 
-#### Play and pause the video {/*play-and-pause-the-video*/}
+#### वीडियो को प्ले और पॉज करें
```
Comment (arshadkazmi42) [inline src/content/learn/manipulating-the-dom-with-refs.md:662]:
```suggestion
#### वीडियो को प्ले और पॉज करें {/*play-and-pause-the-video*/}
```

File: src/content/learn/manipulating-the-dom-with-refs.md [hunk 14]
```diff
@@ -749,13 +751,13 @@ button { display: block; margin-bottom: 20px; }
 
 </Sandpack>
 
-In order to handle the built-in browser controls, you can add `onPlay` and `onPause` handlers to the `<video>` element and call `setIsPlaying` from them. This way, if the user plays the video using the browser controls, the state will adjust accordingly.
+ब्राउज़र के बिल्ट-इन कंट्रोल्स को हैंडल करने के लिए, आप `<video>` तत्व पर `onPlay` और `onPause` हैंडलर्स जोड़ सकते हैं और इनसे `setIsPlaying` को कॉल कर सकते हैं। इस तरह, अगर उपयोगकर्ता ब्राउज़र कंट्रोल्स का उपयोग करके वीडियो प्ले करता है, तो राज्य (state) इसके अनुसार समायोजित हो जाएगा।
 
 </Solution>
 
-#### Focus the search field {/*focus-the-search-field*/}
+#### खोज फ़ील्ड को फ़ोकस करें
```
Comment (arshadkazmi42) [inline src/content/learn/manipulating-the-dom-with-refs.md:758]:
```suggestion
#### खोज फ़ील्ड को फ़ोकस करें {/*focus-the-search-field*/}
```

File: src/content/learn/manipulating-the-dom-with-refs.md [hunk 15]
```diff
@@ -817,9 +819,9 @@ button { display: block; margin-bottom: 10px; }
 
 </Solution>
 
-#### Scrolling an image carousel {/*scrolling-an-image-carousel*/}
+#### इमेज कैरोसेल को स्क्रॉल करना
```
Comment (arshadkazmi42) [inline src/content/learn/manipulating-the-dom-with-refs.md:822]:
```suggestion
#### इमेज कैरोसेल को स्क्रॉल करना {/*scrolling-an-image-carousel*/}
```

File: src/content/learn/manipulating-the-dom-with-refs.md [hunk 16]
```diff
@@ -1043,13 +1044,13 @@ img {
 
 </Solution>
 
-#### Focus the search field with separate components {/*focus-the-search-field-with-separate-components*/}
+#### अलग-अलग घटकों के साथ सर्च फील्ड पर फोकस करें
```
Comment (arshadkazmi42) [inline src/content/learn/manipulating-the-dom-with-refs.md:1047]:
```suggestion
#### अलग-अलग घटकों के साथ सर्च फील्ड पर फोकस करें {/*focus-the-search-field-with-separate-components*/}
```

**Review by arshadkazmi42 — CHANGES_REQUESTED** [review]
I just ran the CI and it was throwing the same error. I did some debugging and found there were few heading links were removed which was causing the issue and that is the reason you were not able to commit.

I have left the comments to fix those. please double check if another heading is missing that.

File: src/content/learn/manipulating-the-dom-with-refs.md (hunk 12 above)
Comment (MadDevMax) [reply src/content/learn/manipulating-the-dom-with-refs.md]: reverted

File: src/content/learn/manipulating-the-dom-with-refs.md (hunk 14 above)
Comment (MadDevMax) [reply src/content/learn/manipulating-the-dom-with-refs.md]: added back

File: src/content/learn/manipulating-the-dom-with-refs.md (hunk 15 above)
Comment (MadDevMax) [reply src/content/learn/manipulating-the-dom-with-refs.md]: added back

File: src/content/learn/manipulating-the-dom-with-refs.md (hunk 16 above)
Comment (MadDevMax) [reply src/content/learn/manipulating-the-dom-with-refs.md]: added back

**Comment by MadDevMax** [comment]
@arshadkazmi42 heading issues are fixed now hopefully 🥲

**Review by arshadkazmi42 — APPROVED** [review]
Looks good 👍

**Pull request merged**